multinet/api/views/alttxt.py

```diff
@@ -0,0 +1,77 @@
+from typing import Union
+import json
+
+from rest_framework.viewsets import ReadOnlyModelViewSet
+from django.http import JsonResponse
+from rest_framework.exceptions import ValidationError, ParseError
+from rest_framework import status, serializers
+from django.core.files.uploadedfile import UploadedFile
+
+from alttxt.enums import Level, Verbosity, Explanation, AggregateBy
+from alttxt.generator import AltTxtGen
+from alttxt.models import DataModel, GrammarModel
+from alttxt.parser import Parser
+from alttxt.tokenmap import TokenMap
+
+class AlttxtSerializer(serializers.Serializer):
+    verbosity = serializers.ChoiceField(choices=Verbosity.list())
+    level = serializers.ChoiceField(choices=Level.list())
+    explain = serializers.ChoiceField(choices=Explanation.list())
+    title = serializers.CharField(max_length=200, default="")
+    data = serializers.FileField(required=True)
+
+class AlttxtQueryViewSet(ReadOnlyModelViewSet):
+    authentication_classes = []
+    permission_classes = []
+
+    def post(self, request) -> JsonResponse:
+        """
+        Process and respond to a form-multipart request
+        containing the params and data needed to generate an alttxt
+        """
+
+        # Get data and serialize it
+        serializer = AlttxtSerializer(data=request.data)
+        if not serializer.is_valid():
+            raise ValidationError(serializer.errors)
```

Comment:
This change isn't quite what I was thinking. I was thinking more along the lines of 
```suggestion
            return HttpResponseBadRequest('Invalid data file: JSON must not be aggregated')
```

Please change this for all instances of this

Reply:
My bad, I thought the message passed to the error would be included in the response- I'll switch them